pkg/instrumentation/podmutator.go

```diff
@@ -42,14 +44,102 @@ type instPodMutator struct {
 	Recorder    record.EventRecorder
 }
 
+type instrumentationWithContainers struct {
+	Instrumentation *v1alpha1.Instrumentation
+	Containers      string
+}
+
 type languageInstrumentations struct {
-	Java        *v1alpha1.Instrumentation
-	NodeJS      *v1alpha1.Instrumentation
-	Python      *v1alpha1.Instrumentation
-	DotNet      *v1alpha1.Instrumentation
-	ApacheHttpd *v1alpha1.Instrumentation
-	Go          *v1alpha1.Instrumentation
-	Sdk         *v1alpha1.Instrumentation
+	Java        instrumentationWithContainers
+	NodeJS      instrumentationWithContainers
+	Python      instrumentationWithContainers
+	DotNet      instrumentationWithContainers
+	ApacheHttpd instrumentationWithContainers
+	Go          instrumentationWithContainers
+	Sdk         instrumentationWithContainers
+}
+
+// Check if single instrumentation is configured for Pod and return which is configured.
+func (langInsts languageInstrumentations) isSingleInstrumentationEnabled() (bool, string) {
+	// Check if more than one field is not nil
+	count := 0
+	enabledInstrumentation := ""
+	value := reflect.ValueOf(langInsts)
+	for i := 0; i < value.NumField(); i++ {
+		field := value.Field(i).FieldByName("Instrumentation")
+		if !field.IsNil() {
+			count++
+			enabledInstrumentation = value.Type().Field(i).Name
+		}
+	}
```

Comment:
Do we have to use reflection?  Can we do `langInsts.Java.Instrumentation == nil` and repeat for each language?  Avoiding reflection will make these functions easier to read/comprehend.

Reply:
Sure, I've decided to use reflection to generate less lines of code, but you're right without reflection it's easier to understand what's going on there :)